yt_dlp/extractor/youtube.py

```diff
@@ -688,26 +727,46 @@ def _extract_session_index(*data):
             if session_index is not None:
                 return session_index
 
-    def _data_sync_id_to_delegated_session_id(self, data_sync_id):
+    @staticmethod
+    def _parse_data_sync_id(data_sync_id):
+        """
+        Parse data_sync_id into delegated_session_id and user_session_id.
+
+        data_sync_id is of the form "delegated_session_id||user_session_id" for secondary channel
+        and just "user_session_id||" for primary channel.
+
+        @param data_sync_id: data_sync_id string
+        @return: Tuple of (delegated_session_id, user_session_id)
+        """
         if not data_sync_id:
-            return
-        # datasyncid is of the form "channel_syncid||user_syncid" for secondary channel
-        # and just "user_syncid||" for primary channel. We only want the channel_syncid
-        channel_syncid, _, user_syncid = data_sync_id.partition('||')
-        if user_syncid:
-            return channel_syncid
+            return None, None
+        first, _, second = data_sync_id.partition('||')
+        return first if second else None, second if second else first
```

Comment:
```suggestion
        if second:
            return first, second
        return None, first
```
This is easier to read imo